lib/puppet/defaults.rb

```diff
@@ -854,11 +854,23 @@ def self.default_diffargs
         This is distinct from the certificate authority's CRL."
     },
     :certificate_revocation => {
-        :default  => true,
-        :type     => :boolean,
-        :desc     => "Whether certificate revocation should be supported by downloading a
-          Certificate Revocation List (CRL)
-          to all clients.  If enabled, CA chaining will almost definitely not work.",
+        :default  => 'chain',
+        :type     => :certificate_revocation,
+        :desc     => <<EOT
+Whether certificate revocation checking should be enabled, and what level of checking should be performed.
+
+When certificate_revocation is set to 'true' or 'chain', Puppet will download the CA CRL and will perform revocation
+checking against each certificate in the chain. When Puppet is using a single root certificate and has no intermediates
```

Comment:
This is that world; unfortunately that world is still broken because we can't load multiple CRLs. My intention with this was to maintain the current behavior to reduce surprises, and when we can support multiple CRLs the chain cert verification will start working. This is a small amount of paranoia and a larger amount of future proofing.